Automattic/jetpack
# Update sharing block controls

Fixes https://github.com/Automattic/jetpack/issues/34115

## Proposed changes:

This is follow up to https://github.com/Automattic/jetpack/pull/34181 so that buttons will have working logic. 
For now, we only using the most important part of redirects.
- Copied `sharing-sources-block` to the block folder leaving only logic inside (no html and styles).
- Js was moved to `view.js`
 

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






- Spin up your site with local changes
- Make sure you have beta block enabled. In your environment (example Docker) wp-config.php you should have define( 'JETPACK_BLOCKS_VARIATION', 'beta' );
- Add Sharing Buttons block to a post and check it's behaviour.
- View the post and check Sharing Buttons functionality.
